### PR TITLE
chore: update TS to version 5.3.2 and adjust config []

### PR DESCRIPTION
**Pull request by Chaoste**
In the TS guild, we discussed that we want to bring most of our packages to TS 5.

### CODE REVIEW

File: packages/rich-text-from-markdown/package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "rollup-plugin-sourcemaps": "^0.6.3",
     "rollup-plugin-typescript2": "^0.36.0",
     "ts-jest": "^29.1.2",
-    "tslib": "^2.3.1",
+    "tslib": "^2.6.2",
     "tslint": "^6.1.3",
     "typescript": "^4.4.2"
```
Comment (andipaetzold) [inline packages/rich-text-from-markdown/package.json:58]: Why not updating these packages as well?

Comment (Chaoste) [reply packages/rich-text-from-markdown/package.json:58]: Upps, on it!

File: packages/rich-text-from-markdown/package.json [hunk 2]
```diff
@@ -53,9 +53,9 @@
     "rollup-plugin-sourcemaps": "^0.6.3",
     "rollup-plugin-typescript2": "^0.36.0",
     "ts-jest": "^29.1.2",
-    "tslib": "^2.3.1",
+    "tslib": "^2.6.2",
```
Comment (andipaetzold) [inline packages/rich-text-from-markdown/package.json:56]: `tslib` should be a regular dep, not `devDep`. But it's not used anyway, so we should be able to remove it

File: tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "moduleResolution": "node",
-    "target": "es5",
+    "target": "es6",
```
Comment (andipaetzold) [inline tsconfig.json:4]: Why this change? Could we replace with https://github.com/tsconfig/bases instead of creating a custom tsconfig?

In any case, this should trigger a release, right?

**Review by andipaetzold — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 3 above)
Comment (Chaoste) [reply tsconfig.json]: I found an issue during build time that according to some issues was fixed by that. I just rolled back to `es5` and don't see it anymore happening. Maybe it was an issue of not using the same TS version in all packages simultaneously. So you're suggesting installing and adding `"extends": "@tsconfig/recommended/tsconfig.json"` to the TSconfig?

Comment (andipaetzold) [reply tsconfig.json]: Yep. Of course only if it doesn't require any more work or changes in the code.

**Review by andipaetzold — APPROVED** [review]
